src/Transit/Internal/Pipeline.hs

```diff
@@ -0,0 +1,132 @@
+module Transit.Internal.Pipeline
+  ( sendPipeline
+  , receivePipeline
+  )
+where
+
+import Protolude
+
+import Crypto.Hash (SHA256(..))
+import Data.Conduit ((.|))
+import Data.ByteString.Builder(toLazyByteString, word32BE)
+import Data.Binary.Get (getWord32be, runGet)
+import Crypto.Saltine.Internal.ByteSizes (boxNonce)
+import System.FilePath ((</>))
+
+import qualified Crypto.Hash as Hash
+import qualified Conduit as C
+import qualified Data.Conduit.Network as CN
+import qualified Data.Conduit.Binary as CB
+import qualified Data.Binary.Builder as BB
+import qualified Data.ByteString as BS
+import qualified Data.ByteString.Lazy as BL
+import qualified Crypto.Saltine.Core.SecretBox as SecretBox
+import qualified Crypto.Saltine.Class as Saltine
+
+import Transit.Internal.Network
+import Transit.Internal.Crypto
+
+-- | Given the peer network socket and the file path to be sent, this Conduit
+-- pipeline reads the file, encrypts and send it over the network. A sha256
+-- sum is calculated on the input before encryption to compare with the
+-- receiver's decrypted copy.
+sendPipeline :: C.MonadResource m =>
+                FilePath
+             -> TCPEndpoint
+             -> SecretBox.Key
+             -> C.ConduitM a c m (Text, ())
+sendPipeline fp (TCPEndpoint s) key =
+  C.sourceFile fp .| sha256PassThroughC `C.fuseBoth` (encryptC key .| CN.sinkSocket s)
+
+-- | Receive the encrypted bytestream from a network socket, decrypt it and
+-- write it into a file, also calculating the sha256 sum of the decrypted
+-- output along the way.
+receivePipeline :: C.MonadResource m =>
+                   FilePath
+                -> Int
+                -> TCPEndpoint
+                -> SecretBox.Key
+                -> C.ConduitM a c m (Text, ())
+receivePipeline fp len (TCPEndpoint s) key =
+    CN.sourceSocket s
+    .| assembleRecordC
+    .| decryptC key
+    .| CB.isolate len
+    .| sha256PassThroughC `C.fuseBoth` C.sinkFileCautious ("./" </> fp)
```

Comment:
Wow! Thanks. I didn't think about that at all and totally missed it. We only send the file name to the other side, not the path because at the destination, the path would not make sense. So, `fp` (standing for filepath) is a misnomer. I will correct that as well.

There are some packages available that allows one to find the current working directory. Trying to see if I can avoid pulling additional dependencies and manage it.